src/background/listener/bee-api.listener.ts

```diff
@@ -0,0 +1,85 @@
+import { StoreObserver, getItem } from '../../utils/storage'
+export class BeeApiListener {
+  private beeApiUrl: string
+
+  public constructor(private storeObserver: StoreObserver) {
+    this.beeApiUrl = 'http://localhost:1633'
+    this.addStoreListeners()
+    this.asyncInit()
+    this.addBzzListeners()
+  }
+
+  private addBzzListeners() {
+    // typed in 'bzz://{content-address}' URI in search bar triggers redirect to gateway BZZ address
+    // only react if google search has been set for searching
+    chrome.webRequest.onBeforeRequest.addListener(
+      (details: chrome.webRequest.WebRequestBodyDetails) => {
+        console.log('Original BZZ Url', details.url)
+        const urlParams = new URLSearchParams(details.url)
+        for (const e of urlParams.entries()) {
+          console.log('entries', e)
+        }
+        const query = urlParams.get('oq')
+
+        if (!query || !query.startsWith('bzz://')) return
+
+        console.log('bzz address', query)
+        console.log('redirect to', `${this.beeApiUrl}/bzz/${query.substr(6)}`)
+
+        return {
+          redirectUrl: `${this.beeApiUrl}/bzz/${query.substr(6)}`,
+        }
+      },
+      {
+        urls: ['https://www.google.com/search?*'],
+      },
+      ['blocking'],
+    )
+
+    //redirect from web+bzz custom protocol
+    chrome.webRequest.onBeforeRequest.addListener(
+      details => {
+        const urlArray = details.url.split('web%2Bbzz%3A%2F%2F')
+
+        // no match
+        if (urlArray.length === 1) return
+
+        const redirectUrl = `${this.beeApiUrl}/bzz/${urlArray[1]}`
+        console.log(`BZZ redirect to ${redirectUrl} from ${details.url}`)
+
+        return { redirectUrl }
+      },
+      { urls: [`${this.beeApiUrl}/dapp-request?bzz-address=*`] },
+      ['blocking'],
+    )
+
+    // Given value of 'bzz-resource' is a bare bzz address without any prefix
+    // redirect by injected script to resources which does not handle custom protocols by default
```

Comment:
Sorry I really don't understand this sentence. Can you elaborate a bit? I'd be glad to help you summarize it.

Reply:
`bzz-resource` is a query field in the fake URL and its value doesn't have any prefix like `web+bzz://` nor `bzz://`, just the BZZ endpoint reference, that does not need any string manipulation.
The second row tells, this redirection is only a mapping that a custom protocol handler could solve by itself, but the injected script's initializer does instead.

Reply:
Does this make sense?
```suggestion
    // Redirects 'bzz-resource=<content_hash>'  to '<gateway>/bzz/<content_hash>
    // Used to load page resources like images
```